common/usermode.c

```diff
@@ -125,6 +158,52 @@ static inline long __user_text syscall(long syscall_nr, long arg1, long arg2, lo
     return return_code;
 }
 
+static inline long __user_text _sysenter(long syscall_nr, long arg1, long arg2, long arg3,
+                                         long arg4, long arg5) {
+    register long return_code asm(STR(_ASM_AX));
+    register long _arg4 asm("r8") = arg4;
+    register long _arg5 asm("r9") = arg5;
+    register long _arg3 asm("r10") = arg3;
+
+    /* clang-format off */
+    asm volatile (
+        "mov %%" STR(_ASM_SP) ", %%" STR(_ASM_CX) "\n"
+        "lea 1f(%%"STR(_ASM_IP)"), %%" STR(_ASM_DX) "\n"
+        "sysenter\n"
+        "1: "
+        : "=a"(return_code)
+        : "0"(syscall_nr), "S"(arg1), "D" (arg2), "r" (_arg3), "r"(_arg4), "r"(_arg5)
+        : STR(_ASM_CX), STR(_ASM_DX)
+    );
+    /* clang-format on */
+
+    return return_code;
+}
+
+static __user_bss enum syscall_mode sc_mode = SYSCALL_MODE_SYSCALL;
```

Comment:
`__user_data`, as it gets initialized (admitted, the value is `0`, but lets not be overly cleaver here).